test/test_cli_integration.py

```diff
@@ -166,7 +176,64 @@ def test_https_cli_can_set_and_get_a_defined_variable(self):
             **self.HTTPS_CA_BUNDLE_ENV_VAR
         })
 
+        self.set_variable(CliIntegrationTest.DEFINED_VARIABLE2, "r")  # TODO: value should be special chars and it fails "!@#$%asdfgh&*()g,lpokmnjiub")
```

Comment:
if you are repeating yourself you can take the logic into a separate function and pass in arguments for the variable values. 
This probably already exists...